apps/smbmount.sh

```diff
@@ -289,6 +289,7 @@ then
             msg_box "Your deletion of /mnt/smbshares/3 was successfull!"
         fi
     fi
+    run_app_script smbmount
```

Comment:
I don't understand this. Are you running the script itself when everything is done?

Reply:
> I don't understand this. Are you running the script itself when everything is done?

Yes. The idea is to have a kind of mainmenu for managing smb-mounts.
From there you can mount, unmount, add list and delete smb-shares.
exiting only works from the mainmenu.

Reply:
Hmm... OK....

Maybe I'm just tired now, but what would happen is:

1.  Run the script
2. Make your choices
3. Run the script again inside the same script that is already running 
4. (endless loop)

It doesn't make sense to me.

Reply:
no, because if you exit from the main menu, you exit the whole script.
The main menu is just a whiptile list where you can choose an option. You should definitely try it yourself. I know the code doesn't look that nice, and i have to refactor a lot of things but it works quiet well!

Reply:
![image](https://user-images.githubusercontent.com/42591237/72194213-1df47280-340d-11ea-93d4-1ef9fbec674e.png)
this is the main menu

Reply:
actually the best part of this script imo is that new smbshares get automatically mounted to the Nextcloud-admin group, so that you are instantly able to access files in that share or just manage it to make it available to other users or just edit the path.

Reply:
> Hmm... OK....
> 
> Maybe I'm just tired now, but what would happen is:
> 
> 1. Run the script
> 2. Make your choices
> 3. Run the script again inside the same script that is already running
> 4. (endless loop)
> 
> It doesn't make sense to me.

okay, you are right. do you have another idea how to go back to the main menu after done with a part? or should it just exit the script?

Reply:
> > Hmm... OK....
> > Maybe I'm just tired now, but what would happen is:
> > 
> > 1. Run the script
> > 2. Make your choices
> > 3. Run the script again inside the same script that is already running
> > 4. (endless loop)
> > 
> > It doesn't make sense to me.
> 
> okay, you are right. do you have another idea how to go back to the main menu after done with a part? or should it just exit the script? or do I have to put every option in different files to be able to do that? (probably an advantege: maybe the script gets slicker then.)

I found a possibility: use exec to start the same script again and end the last execution.

Reply:
But exec would probably also exit the startup-script. Wouldn't it?

Reply:
> But exec would probably also exit the startup-script. Wouldn't it?

That's the thing with developing, you need to test test test test. :)

I just want to merge the TLS PR first. Will give this my attention once that's done.

One obvious thing I saw is that you can make a lot of repeating code functions instead. Teyon't need to be put in `lib.sh` as they only are used here.

```
mount_in_files_external_app() {
occ_command app:enable files_external
            MOUNT_ID=$(occ_command files_external:create $1 local null::null -c datadir=$2)
            MOUNT_ID=${MOUNT_ID//[!0-9]/}
            occ_command files_external:applicable --add-group=admin "$MOUNT_ID" -q
            msg_box "Your mount was successfull, congratulations!\n It is accessible in your root directory in $2.\nYou can now use the Nextcloud external storage app to access files there."
}

mount_in_files_external_app SMB1 /mnt/smbshares/1

```

It's simple really, just make repeating code a function and call $1 for the first line after the function, and $2 for the second word in the function, then $3, $4 and so on if it's a larger function.

Reply:
thanks for the example!
But I think loops fit better here since every function is just needed here.
So I've put everything possible into loops.
but it would make sense think about putting some of the remaining code into functions.

Reply:
> But exec would probably also exit the startup-script. Wouldn't it?

@enoch85 Since that would probably be a problem, I could maybe solve it by putting every option into a new file and put the main script into a loop. What do you think about this?

Reply:
This is done.